### ISSUE TITLE
Mac non executable app / cannot launch

### ISSUE THREAD

**Issue by mtsamis** [libgdx/packr#94]
Hallo,

I'm using packr to pack a jar for osx but can't get it working. Im using the equivalent config with  win/linux which work but the resulting osx program isn't working. I tried with a minimal java program (no dependancies etc) and it still didnt work. When i manually launch the executable in Contents/MacOS/ i get up a terminal which goes until "Invoking static mainPackage.AppMain.main() function..." and then nothing happens, it stops there and i can only force terminate the program. (if I add .app extension to the Contents folder and then launch the MacOS/ executable i get "Error: failed to load configuration: config.json"). Otherwise for win/linux packr performed great and is a really handy and flexible packaging tool.

Any help is deeply appreciated, thanks

Macbook air - version 10.10.5

**Comment by jammehcow** [comment]
I'm also suffering from this issue. I'm running Sierra 10.12.3 so I thought it might've been just that release, but nice to know that it's not I guess. This is a pain in the ass for those wanting to distribute to OS X so I think it should be a high priority if this project is even still active.

**Comment by code-disaster** [comment]
Without more information, my best guess is to try a different JVM. For example, right now we use a Azul Zulu JVM with deployments of our Java 8 game in development, and it's working pretty well so far. (but has a minimum requirement of OS X 10.9)

**Comment by anubitum** [comment]
I had the same problem.

After trying different JVM (different version JDK, Zulu), I tried removing the VM arguments ("Xmx1G" and "XstartOnFirstThread") and it worked. Try to play with config.json "vmArgs", directly into .app package to see if it works for you too.

**Comment by raincole** [comment]
I got the same issue. Removing `XstartOnFirstThread` fixed it, but it makes Packr unusable for LWJGL3 based app.

**Comment by sco-tt** [comment]
Just wanted to add that switching to the Zulu JVM solved this for me.

**Comment by neomatrix369** [comment]
I'm trying this out now, will keep the thread posted of my own experience. Rebuild Packr version 2.1

**Comment by neomatrix369** [comment]
I removed the vm args flag and switched to Zulu MacOS JDK 1.8 (8u192) but it still does the same, hangs and does nothing:

```
➜ $ ./record_screen_and_upload.sh   
record folder already exists, proceding
INFO  [main]       - Starting recording app
```

**Comment by neomatrix369** [comment]
I ran it with the -verbose option:
```
[snipped]
[Loaded javax.management.InstanceNotFoundException from /path/to/Contents/Resources/jre/lib/rt.jar]
[Loaded javax.management.MBeanServerFactory from /path/to/Contents/Resources/jre/lib/rt.jar]
[Loaded java.lang.management.ManagementFactory from /path/to/Contents/Resources/jre/lib/rt.jar]
[Loaded java.util.IdentityHashMap$KeySet from /path/to/Contents/Resources/jre/lib/rt.jar]
[Loaded java.util.IdentityHashMap$IdentityHashMapIterator from /path/to/Contents/Resources/jre/lib/rt.jar]
[Loaded java.util.IdentityHashMap$KeyIterator from /path/to/Contents/Resources/jre/lib/rt.jar]
```

**Comment by code-disaster** [comment]
I doubt that "Java Management Extensions" is shipped as part of the Zulu JDK.

**Comment by neomatrix369** [comment]
I have fixed the issue, it was something silly on my part - I used the minimiseJre option and it would strip off things that were needed during execution, removing that option the issue was fixed.

Let me try again from scratch and see what happens.

**Comment by neomatrix369** [comment]
I works now from scratch - so its missing JDK/JRE dependencies which causes it to freeze. Its now all sorted for me.
